reframe/frontend/statistics.py

```diff
@@ -171,7 +171,10 @@ def json(self, force=False):
                     entry['result'] = 'success'
                     entry['outputdir'] = check.outputdir
 
-                if hasattr(check, 'perf_patterns'):
+                if (
+                    hasattr(check, 'perf_variables') or
+                    hasattr(check, 'perf_paterns')
+                ):
```

Comment:
I think this might just do the trick:
```suggestion
                if check.perfvalues:
```